WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+                
+                if changesSaved {
+                    self?.setFilterWithPostStatus(viewController.post.status)
+                }
+                
+                viewController.navigationController?.popViewControllerAnimated(true)
+            }
+            
+            postViewController.hidesBottomBarWhenPushed = true
+            navigationController?.pushViewController(postViewController, animated: true)
+        } else {
+            // In legacy mode, view means edit
+            let editPostViewController = WPLegacyEditPostViewController(post: apost)
+            let navController = UINavigationController(rootViewController: editPostViewController)
+            navController.modalPresentationStyle = .FullScreen
+            navController.restorationIdentifier = WPLegacyEditorNavigationRestorationID
+            navController.restorationClass = WPLegacyEditPostViewController.self
+            
+            presentViewController(navController, animated: true, completion: nil)
+        }
+    }
+    
+    private func promptThatPostRestoredToFilter(filter: PostListFilter) {
+        var message = NSLocalizedString("Post Restored to Drafts", comment: "Prompts the user that a restored post was moved to the drafts list.")
+        
+        switch filter.filterType {
+        case .Published:
+            message = NSLocalizedString("Post Restored to Published", comment: "Prompts the user that a restored post was moved to the published list.")
+            break
+        case .Scheduled:
+            message = NSLocalizedString("Post Restored to Scheduled", comment: "Prompts the user that a restored post was moved to the scheduled list.")
+            break
+        default:
+            break
+        }
+        
+        let alertCancel = NSLocalizedString("OK", comment: "Title of an OK button. Pressing the button acknowledges and dismisses a prompt.")
+        
+        let alertController = UIAlertController(title: nil, message: message, preferredStyle: .Alert)
+        alertController.addCancelActionWithTitle(alertCancel, handler: nil)
+        alertController.presentFromRootViewController()
+    }
+    
+    private func viewStatsForPost(apost: AbstractPost) {
+        // Check the blog
+        let blog = apost.blog
+        
+        if blog.supports(.Stats) {
+            // Needs Jetpack.
+            return
+        }
+        
+        WPAnalytics.track(.PostListStatsAction, withProperties: propertiesForAnalytics())
+        
+        // Push the Stats Post Details ViewController
+        let identifier = NSStringFromClass(StatsPostDetailsTableViewController.self)
+        let service = BlogService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        let statsBundle = NSBundle(forClass: WPStatsViewController.self)
+        let path = statsBundle.pathForResource("WordPressCom-Stats-iOS", ofType: "bundle")!
```

Comment:
It would be nice to avoid explicitly unwrapped optionals where ever possible. If for no other reason than paranoia. This way if a glitch is accidentally introduced in a future version of iOS or of the app we might avoid crashing on an unexpected nil.
Maybe change to something like 
```
        guard let path = statsBundle.pathForResource("WordPressCom-Stats-iOS", ofType: "bundle") else {
            return
        }
        let bundle = NSBundle(path: path)
        let statsStoryboard = UIStoryboard(name: self.dynamicType.statsStoryboardName, bundle: bundle)
        if let controller = statsStoryboard.instantiateViewControllerWithIdentifier(identifier) as? StatsPostDetailsTableViewController {
        
            controller.postID = apost.postID
            controller.postTitle = apost.titleForDisplay()
            controller.statsService = WPStatsService(siteId: blog.dotComID, siteTimeZone: service.timeZoneForBlog(blog), oauth2Token: blog.authToken, andCacheExpirationInterval: self.dynamicType.statsCacheInterval)
            
            navigationController?.pushViewController(controller, animated: true)
        }
```